server/src/main/java/org/elasticsearch/health/SimpleHealthIndicatorDetails.java

```diff
@@ -26,4 +26,21 @@ public SimpleHealthIndicatorDetails(Map<String, Object> details) {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.map(details);
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Consider turning `SimpleHealthIndicatorDetails` into a record, so we don't need to implement `equals&hashCode`